### PR TITLE
v7: hoist devDependencies

### PR DESCRIPTION
**Pull request by djskinner**
Move common devDependencies to the root of the lerna repo. Moved dependencies have been updated. Note that devDependencies providing "binary" executables have not been hoisted. 

Reference: https://github.com/lerna/lerna/blob/master/README.md#common-devdependencies

### CODE REVIEW

File: packages/plugin-expo-app/package-lock.json [hunk 1]
```diff
@@ -101,28 +101,6 @@
...
-		"@bugsnag/cuid": {
-			"version": "3.0.0",
-			"resolved": "https://registry.npmjs.org/@bugsnag/cuid/-/cuid-3.0.0.tgz",
-			"integrity": "sha512-LOt8aaBI+KvOQGneBtpuCz3YqzyEAehd1f3nC5yr9TIYW1+IzYKa2xWS4EiMz5pPOnRPHkyyS5t/wmSmN51Gjg=="
-		},
-		"@bugsnag/safe-json-stringify": {
-			"version": "4.0.0",
-			"resolved": "https://registry.npmjs.org/@bugsnag/safe-json-stringify/-/safe-json-stringify-4.0.0.tgz",
-			"integrity": "sha512-eb0yG+PNTdpRXIhfEq4IFi2wBfpE1DjrAlkyehinu9RxE1PKteqJioLmfQjvkG3UviGu/k5pcamYMxNuCAMIoQ=="
-		},
```
Comment (djskinner) [inline packages/plugin-expo-app/package-lock.json]: These lines were removed by lerna bootstrap. I believe this change is correct as the changed packages now reflect the other packages that do not contain these entries in their package-lock.json which makes things more consistent.

File: packages/plugin-expo-device/package-lock.json [hunk 2]
```diff
@@ -101,28 +101,6 @@
...
-		"@bugsnag/cuid": {
-			"version": "3.0.0",
-			"resolved": "https://registry.npmjs.org/@bugsnag/cuid/-/cuid-3.0.0.tgz",
-			"integrity": "sha512-LOt8aaBI+KvOQGneBtpuCz3YqzyEAehd1f3nC5yr9TIYW1+IzYKa2xWS4EiMz5pPOnRPHkyyS5t/wmSmN51Gjg=="
-		},
-		"@bugsnag/safe-json-stringify": {
-			"version": "4.0.0",
-			"resolved": "https://registry.npmjs.org/@bugsnag/safe-json-stringify/-/safe-json-stringify-4.0.0.tgz",
-			"integrity": "sha512-eb0yG+PNTdpRXIhfEq4IFi2wBfpE1DjrAlkyehinu9RxE1PKteqJioLmfQjvkG3UviGu/k5pcamYMxNuCAMIoQ=="
-		},
```
Comment (djskinner) [inline packages/plugin-expo-device/package-lock.json]: Also here. See previous comment

File: packages/plugin-react-native-global-error-handler/package-lock.json [hunk 3]
```diff
@@ -101,28 +101,6 @@
...
-		"@bugsnag/cuid": {
-			"version": "3.0.0",
-			"resolved": "https://registry.npmjs.org/@bugsnag/cuid/-/cuid-3.0.0.tgz",
-			"integrity": "sha512-LOt8aaBI+KvOQGneBtpuCz3YqzyEAehd1f3nC5yr9TIYW1+IzYKa2xWS4EiMz5pPOnRPHkyyS5t/wmSmN51Gjg=="
-		},
-		"@bugsnag/safe-json-stringify": {
-			"version": "4.0.0",
-			"resolved": "https://registry.npmjs.org/@bugsnag/safe-json-stringify/-/safe-json-stringify-4.0.0.tgz",
-			"integrity": "sha512-eb0yG+PNTdpRXIhfEq4IFi2wBfpE1DjrAlkyehinu9RxE1PKteqJioLmfQjvkG3UviGu/k5pcamYMxNuCAMIoQ=="
-		},
```
Comment (djskinner) [inline packages/plugin-react-native-global-error-handler/package-lock.json]: Also here. See previous comment

File: packages/plugin-react-native-unhandled-rejection/package-lock.json [hunk 4]
```diff
@@ -101,28 +101,6 @@
...
-		"@bugsnag/cuid": {
-			"version": "3.0.0",
-			"resolved": "https://registry.npmjs.org/@bugsnag/cuid/-/cuid-3.0.0.tgz",
-			"integrity": "sha512-LOt8aaBI+KvOQGneBtpuCz3YqzyEAehd1f3nC5yr9TIYW1+IzYKa2xWS4EiMz5pPOnRPHkyyS5t/wmSmN51Gjg=="
-		},
-		"@bugsnag/safe-json-stringify": {
-			"version": "4.0.0",
-			"resolved": "https://registry.npmjs.org/@bugsnag/safe-json-stringify/-/safe-json-stringify-4.0.0.tgz",
-			"integrity": "sha512-eb0yG+PNTdpRXIhfEq4IFi2wBfpE1DjrAlkyehinu9RxE1PKteqJioLmfQjvkG3UviGu/k5pcamYMxNuCAMIoQ=="
-		},
```
Comment (djskinner) [inline packages/plugin-react-native-unhandled-rejection/package-lock.json]: Also here. See previous comment

File: packages/plugin-expo-app/package-lock.json (hunk 1 above)
Comment (bengourley) [reply packages/plugin-expo-app/package-lock.json]: Yeah it looks like these got a little messed up and stopped linking locally to `@bugsnag/core` and instead pulling that prerelease version down from npm.

I agree these changes seem correct.

**Comment by djskinner** [comment]
> I'm somewhat unsure about hoisting non-common dependencies – i.e `ncp`, `timekeeper` and `rimraf`. If they are only used by one package, is it of benefit to move the dependency up to the top level?

Yeah I hesitated on this one initially but thought it would be worth a discussion. Happy to change the PR to only hoist dependencies that are currently used in > 1 package.

The principal reason I managed to sell myself on the idea was considering what happens when a developer comes to use `timekeeper` in a second package. In all likelihood they would not be aware of the preference to hoist dev dependencies when used in > 1 package. So they add `timekeeper` to package2, likely getting a different version. Hopefully it gets picked up at PR review and the hoisting can be done then (with the associated package-lock churn). If not caught immediately we'll be seeing this "hoist devDependencies" more often once individual cases are discovered. All that assumes they were aware `timekeeper` was being used in the first place (which is less obvious if not stated in the root package.json) and didn't opt to use a different package offering the same functionality because they weren't aware.

So on balance I felt it was simpler, cleaner and clearer to have all devDependencies hoisted where possible rather than having to deal with the "oh, this is now used in more than one package" scenario each time.

Not that it's a convincing argument in and of itself but it's interesting to note that [lerna does it](https://github.com/lerna/lerna/search?q=touch&unscoped_q=touch).

**Pull request merged**